client/components/banner/test/index.jsx

```diff
@@ -1,165 +1,197 @@
 	} );
 
 	test( 'should render two <PlanPrice /> components when there are two prices', () => {
-		const comp = shallow( <Banner { ...props } price={ [ 100, 80 ] } /> );
-		expect( comp.find( PlanPrice ) ).toHaveLength( 2 );
+		render( <Banner { ...props } price={ [ 100, 80 ] } /> );
+		expect( screen.getByText( '100' ) ).toBeVisible();
+		expect( screen.getByText( '80' ) ).toBeVisible();
 	} );
 
 	test( 'should render no <PlanPrice /> components when there are no prices', () => {
-		const comp = shallow( <Banner { ...props } /> );
-		expect( comp.find( PlanPrice ) ).toHaveLength( 0 );
+		const { container } = render( <Banner { ...props } /> );
+		expect( container.getElementsByClassName( 'plan-price__integer' ) ).toHaveLength( 0 );
 	} );
 
 	test( 'should render a .banner__description when description is specified', () => {
-		const comp = shallow( <Banner { ...props } description="test" /> );
-		expect( comp.find( '.banner__description' ) ).toHaveLength( 1 );
+		render( <Banner { ...props } description="test" /> );
+		expect( screen.getByText( 'test' ) ).toBeVisible();
 	} );
 
 	test( 'should not render a .banner__description when description is not specified', () => {
-		const comp = shallow( <Banner { ...props } /> );
-		expect( comp.find( '.banner__description' ) ).toHaveLength( 0 );
+		const { container } = render( <Banner { ...props } /> );
+		expect( container.getElementsByClassName( 'banner__description"' ) ).toHaveLength( 0 );
 	} );
 
 	test( 'should render a .banner__list when list is specified', () => {
-		const comp = shallow( <Banner { ...props } list={ [ 'test1', 'test2' ] } /> );
-		expect( comp.find( '.banner__list' ) ).toHaveLength( 1 );
-		expect( comp.find( '.banner__list li' ) ).toHaveLength( 2 );
-		expect( comp.find( '.banner__list li' ).at( 0 ).text() ).toContain( 'test1' );
-		expect( comp.find( '.banner__list li' ).at( 1 ).text() ).toContain( 'test2' );
+		const { container } = render( <Banner { ...props } list={ [ 'test1', 'test2' ] } /> );
+		expect( container.querySelectorAll( '.banner__list' ) ).toHaveLength( 1 );
+		expect( container.querySelectorAll( '.banner__list li' ) ).toHaveLength( 2 );
+		expect( screen.getByText( 'test1' ) ).toBeVisible();
+		expect( screen.getByText( 'test2' ) ).toBeVisible();
 	} );
 
 	test( 'should not render a .banner__list when description is not specified', () => {
-		const comp = shallow( <Banner { ...props } /> );
-		expect( comp.find( '.banner__list' ) ).toHaveLength( 0 );
+		const { container } = render( <Banner { ...props } /> );
+		expect( container.getElementsByClassName( '.banner__list' ) ).toHaveLength( 0 );
 	} );
 
 	test( 'should record Tracks event when event is specified', () => {
-		const comp = shallow( <Banner { ...props } event="test" /> );
-		expect( comp.find( 'TrackComponentView' ) ).toHaveLength( 1 );
+		renderWithProvider( <Banner { ...props } event="test" /> );
+		expect( recordTracksEvent.mock.calls[ 0 ][ 1 ].cta_name ).toBe( 'test' );
```

Comment:
Could we make this a bit more readable by using `toHaveBeenCalledWith()` instead?

Reply:
The reason we didn't use this initially is because of the other arguments that are being passed through. Is there a way we could check for 'test' without specifying the rest? 

So if we use `toHaveBeenCalledWith()`, we receive: `"calypso_banner_cta_impression", {"cta_feature": undefined, "cta_name": "test", "cta_size": "regular"}`